File: app/src/main/java/com/communityx/adapters/FriendsPagerAdapter.java
package com.communityx.adapters;

import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;
import com.communityx.fragments.FriendsFragment;
import com.communityx.fragments.InvitationFragment;
import com.communityx.fragments.SuggestionFragment;

public class FriendsPagerAdapter extends FragmentPagerAdapter {

    private Fragment[] fragments = {
            new FriendsFragment(),
            new InvitationFragment(),
            new SuggestionFragment()
    };

    //TODO: Hard Coded String
    private String titles[] = {
            "All Friends",
            "Invitations",
            "Suggestions"
    };

    public FriendsPagerAdapter(FragmentManager fm) {
        super(fm);
    }

    @Override
    public Fragment getItem(int i) {
        return fragments[i];
    }

    @Override
    public int getCount() {
        return fragments.length;
    }

    @Nullable
    @Override
    public CharSequence getPageTitle(int position) {
        return titles[position];
    }
}

File: app/src/main/java/com/communityx/adapters/FriendsPagerAdapter.kt
package com.communityx.adapters

import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentPagerAdapter
import com.communityx.fragments.FriendsFragment
import com.communityx.fragments.InvitationFragment
import com.communityx.fragments.SuggestionFragment

class FriendsPagerAdapter(fm: FragmentManager) : FragmentPagerAdapter(fm) {

    private val fragments = arrayOf(FriendsFragment(), InvitationFragment(), SuggestionFragment())

    //TODO: Hard Coded String
    private val titles = arrayOf("All Allies", "Invitations", "Suggestions")

    override fun getItem(i: Int): Fragment {
        return fragments[i]
    }

    override fun getCount(): Int {
        return fragments.size
    }

    override fun getPageTitle(position: Int): CharSequence? {
        return titles[position]
    }
}
